pkg/buildpack/buildpack_test.go

```diff
@@ -571,3 +737,58 @@ type readerBlob struct {
 func (r *readerBlob) Open() (io.ReadCloser, error) {
 	return r.openFn(), nil
 }
+
+type errorBuildModule struct {
+	flattened bool
+}
+
+func (eb *errorBuildModule) Open() (io.ReadCloser, error) {
+	return nil, errors.New("something happened opening the build module")
+}
+
+func (eb *errorBuildModule) Descriptor() buildpack.Descriptor {
+	return nil
+}
+
+func (eb *errorBuildModule) ContainsFlattenedModules() bool {
+	return eb.flattened
+}
+
+type expectedBuildpack struct {
+	id      string
+	version string
+}
+
+func assertBuildpacksToTar(t *testing.T, actual []buildpack.ModuleTar, expected []expectedBuildpack) {
+	t.Helper()
+	for _, expectedBP := range expected {
+		found := false
+		for _, moduleTar := range actual {
+			if expectedBP.id == moduleTar.Info().ID && expectedBP.version == moduleTar.Info().Version {
+				found = true
+			}
+		}
+		h.AssertTrue(t, found)
+	}
+
+	for _, moduleTar := range actual {
+		h.AssertOnTarEntry(t, moduleTar.Path(), fmt.Sprintf("/cnb/buildpacks/%s", moduleTar.Info().ID),
+			h.IsDirectory(),
+		)
```

Comment:
```suggestion
					id := idx + 1
					h.AssertOnTarEntry(t, tarPath.Path(), fmt.Sprintf("/cnb/buildpacks/buildpack-%d-id", id),
						h.IsDirectory(),
					)
```

`moduleTar` is an output of the function being tested, and as such, it isn't safe to use in the expectation. `moduleTar.Info().ID` was hiding the bug mentioned earlier, as it was returning `2` twice.

Reply:
yeah, I wanted to check first the expected ID and Version in line 764. The comparison is made using the fixed expected values